OptimoveSDK/optimove-sdk/src/main/java/com/optimove/android/optimobile/OptimobileHttpClient.java

```diff
@@ -0,0 +1,118 @@
+package com.optimove.android.optimobile;
+
+import android.os.Build;
+import android.util.Base64;
+
+import androidx.annotation.Nullable;
+
+import com.optimove.android.Optimove;
+import com.optimove.android.OptimoveConfig;
+
+import java.io.IOException;
+import java.util.Collections;
+
+import okhttp3.Callback;
+import okhttp3.ConnectionSpec;
+import okhttp3.MediaType;
+import okhttp3.OkHttpClient;
+import okhttp3.Request;
+import okhttp3.RequestBody;
+import okhttp3.Response;
+
+class OptimobileHttpClient {
+
+    private static final Object lock = new Object();
+    private static OptimobileHttpClient instance;
+
+    private final OkHttpClient okHttpClient;
+    private @Nullable String authHeader;
+
+    static OptimobileHttpClient getInstance() {
+        if (instance != null) {
+            return instance;
+        }
+        synchronized (lock) {
+            if (instance == null) {
+                instance = new OptimobileHttpClient();
+            }
+        }
+        return instance;
+    }
+
+    private OptimobileHttpClient() {
+        okHttpClient = this.buildOkHttpClient();
+    }
+
+    private OkHttpClient buildOkHttpClient() {
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M) {
+            return new OkHttpClient();
+        }
+
+        //ciphers available on Android 4.4 have intersections with the approved ones in MODERN_TLS, but the intersections are on bad cipher list, so,
+        //perhaps not supported by CloudFlare. On older devices allow all ciphers
+        ConnectionSpec spec = new ConnectionSpec.Builder(ConnectionSpec.MODERN_TLS)
+                .allEnabledCipherSuites()
+                .build();
+
+        return new OkHttpClient.Builder()
+                .connectionSpecs(Collections.singletonList(spec))
+                .build();
+    }
+
+    Response postSync(String url, String data) throws IOException, Optimobile.PartialInitialisationException {
+        RequestBody body = RequestBody.create(data, MediaType.parse("application/json; charset=utf-8"));
```

Comment:
If we expect to post JSON data, should we take JSONObject/Array in method & stringify internally? (just prevents misalignment of header vs body).

Perhaps future/not strictly necessary.

Reply:
would be better, but how to pass jsonObject _or_  jsonArray? overload?

Reply:
Overloads in Java, yep.